services/src/test/java/org/exoplatform/task/service/TaskStorageTest.java

```diff
@@ -0,0 +1,130 @@
+import org.exoplatform.task.domain.Project;
+import org.exoplatform.task.domain.Status;
+import org.exoplatform.task.domain.Task;
+import org.exoplatform.task.dto.TaskDto;
+import org.exoplatform.task.service.impl.TaskServiceImpl;
+import org.exoplatform.task.storage.ProjectStorage;
+import org.exoplatform.task.storage.TaskStorage;
+import org.exoplatform.task.storage.impl.ProjectStorageImpl;
+import org.exoplatform.task.storage.impl.TaskStorageImpl;
+
+public class TaskStorageTest extends AbstractTest {
+
+  private TaskHandler     tDAO;
+
+  private CommentHandler  cDAO;
+
+  private DAOHandler      daoHandler;
+
+  private final String    username = "root";
+
+  private LabelHandler    labelHandler;
+
+  private ListenerService listenerService;
+
+  private ProjectStorage  projectStorage;
+
+  private UserService     userService;
+
+  private TaskStorage     taskStorage;
+
+  private TaskService     taskService;
+
+  @Before
+  public void setup() {
+    PortalContainer container = PortalContainer.getInstance();
+
+    daoHandler = (DAOHandler) container.getComponentInstanceOfType(DAOHandler.class);
+    tDAO = daoHandler.getTaskHandler();
+    cDAO = daoHandler.getCommentHandler();
+    projectStorage = new ProjectStorageImpl(daoHandler);
+    labelHandler = daoHandler.getLabelHandler();
+    taskStorage = new TaskStorageImpl(daoHandler, userService, projectStorage);
+    taskService = new TaskServiceImpl(taskStorage, daoHandler, listenerService);
+  }
+
+  @After
+  public void tearDown() {
+    for (Task t : tDAO.findAll()) {
+      t.setStatus(null);
+    }
+    tDAO.updateAll(tDAO.findAll());
+    daoHandler.getLabelTaskMappingHandler().deleteAll();
+    daoHandler.getTaskLogHandler().deleteAll();
+    cDAO.deleteAll();
+    tDAO.deleteAll();
+    labelHandler.deleteAll();
+  }
+
+  @Test
+  public void findTaskWithMembership() throws Exception {
+    // Given
+    Project project = new Project();
+    project.setName("Project1");
+
+    project.setParticipator(new HashSet<String>(Arrays.asList("jhon")));
+
+    project = daoHandler.getProjectHandler().create(project);
+    Status status = newStatusInstance("TO DO", 1);
+    status.setProject(project);
+    daoHandler.getStatusHandler().create(status);
+    Task task1 = newTaskInstance("Task 1", "", "root");
+    task1.setStatus(status);
+    tDAO.create(task1);
+    Project project2 = new Project();
+    project2.setName("Project2");
+    project2 = daoHandler.getProjectHandler().create(project2);
+    Status status1 = newStatusInstance("TO DO", 1);
+    status1.setProject(project2);
+    daoHandler.getStatusHandler().create(status1);
+    Task task2 = newTaskInstance("Task 2", "", "root");
+    task2.setStatus(status1);
+    tDAO.create(task2);
+    List<String> memberships = Arrays.asList("jhon");
+
+    // When
+    List<TaskDto> tasks = taskStorage.findTasks("jhon", memberships, "Task", 10);
+
+    // then
+    assertEquals(1, tasks.size());
+
```

Comment:
```suggestion
   // When
    List<TaskDto> tasks = taskStorage.findTasks("root", memberships, "Task", 10);

    // then
    assertEquals(2, tasks.size());
```

Reply:
Why did you change this `assertEquals(1, tasks.size()) --- > assertEquals(2, tasks.size())` ?
The expected result only one task has to be found, because the user "`jhon`" is only a member in `project2`, so the task that will be displayed by search is the (`Task 2`) in `project2`

Reply:
I didnt change the assert, I add new one. 
When you search task for root  you should find 2